src/main/java/io/github/eliux/mega/cmd/ExportInfo.java

```diff
@@ -10,22 +16,44 @@
  */
 public class ExportInfo {
 
+    private static String dateFormat = "EEE, dd MMM yyyy HH:mm:ss Z";
     private static final Pattern LIST_PATTERN =
             Pattern.compile("(?<remotePath>\\S+) \\(.+link: (?<publicLink>http[s]?://mega.nz/\\S*#.+)\\)");
 
+    private static final Pattern EXPORT_PATTERN =
+            Pattern.compile("\\s(?<remotePath>\\S+)(:)\\s(?<publicLink>http[s]?://mega.nz/\\S*#\\S*)( expires at (?<expireDate>.{3}, .{2} .{3} .{4} .{2}:.{2}:.{2} .+))?");
+
     private final String remotePath;
 
     private final String publicLink;
 
+    private Optional<String> expireDate;
+
     public ExportInfo(String remotePath, String publicLink) {
         this.remotePath = remotePath;
         this.publicLink = publicLink;
     }
 
     public static ExportInfo parseExportInfo(String exportInfoStr) {
-        final String[] tokens = exportInfoStr.replace("Exported ", "")
-                .split(": ");
-        return new ExportInfo(tokens[0], tokens[1]);
+        final Matcher matcher = EXPORT_PATTERN.matcher(exportInfoStr);
+
+        if (matcher.find()) {
+            String remotePath = matcher.group("remotePath");
+            String publicLink = matcher.group("publicLink");
+            String expireDate = matcher.group("expireDate");
+
+            if (expireDate == null && exportInfoStr.contains("expires at")) {
+                throw new MegaInvalidExpireDateException(exportInfoStr);
+            }
+            if (expireDate != null) {
+                return new ExportInfo(remotePath, publicLink)
+                        .setExpireDate(expireDate);
+            }  else {
+                return new ExportInfo(remotePath, publicLink);
+            }
+        }
+
+        throw new MegaInvalidResponseException(exportInfoStr);
```

Comment:
i think it's still necessary. if the "exportInfoStr" is completly different from what it's supposed to be, the `matcher.find()` will be false. So we need to throw en exception.

Reply:
@EliuX

Reply:
I still you to use this recommendation 
https://github.com/EliuX/MEGAcmd4J/pull/27/files/972fff2f63b455a2c8683ad230aabfb79bea9857#r471151356

A `null` checking is a code-smell. You should use an `Optional` object if something is nullable.
In addition, the Exception should have a verbose and clean error message: The type of the exception is important for handling the error semantically, but the error message is what you show to the user to understand what is actually happening based on the context. So, IMHO, a cleaner solution would be:
```
if (matcher.find()) {
  String remotePath = matcher.group("remotePath");
  String publicLink = matcher.group("publicLink");
  String expireDateStr = matcher.group("expireDate");

  final ExportInfo result = new ExportInfo(remotePath, publicLink);
  Optional.ofNullable(expireDateStr)
               .map(MegaUtils::parseFullDate)
               .ifPresent(result::setExpireDate);

  return result;
}

throw new MegaInvalidResponseException('Unexpected export info format');
```

The line `.map(MegaUtils::parseFullDate)` is very important, because it can be used by other functions (DRY) which require to parse full dates as this one: convert a parsed string into a date based on the format we expect. If something is wrong with that date format, then an automated test shall fail in the CI so it can be fixed.

Thanks to the fact that we moved `DATE_FORMAT` and its logic to `MegaUtils#parseFullDate` to validate the format of the dates given by MegaCMD, we can and should skip the format validation, i.e. `.{3}, .{2} .{3} .{4} .{2}:.{2}:.{2} .+` , from the regexp. Otherwise you would be validating this format twice, which is somewhat inconsistent and prone to errors.

Does it look cleaner?